main/config_nodes.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) The Libre Solar Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CONFIG_NODES_H_
+#define CONFIG_NODES_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "ts_client.h"
+
+
+#define CONFIG_LIBRE_SOLAR_TYPE_ID 9
+
+/*
+ * Partition name
+ */
+#define PARTITION "config"
+#define NAMESPACE "main"
+
+
+/*
+ * Categories / first layer node IDs
+ */
+#define ID_ROOT     0x00
+#define ID_INFO     0x18        // read-only device information (e.g. manufacturer, device ID)
+#define ID_CONF     0x30        // configurable settings
+#define ID_CONF_GENERAL 0x31
+#define ID_CONF_EMONCMS 0x37
+#define ID_CONF_MQTT    0x40
+#define ID_INPUT    0x60        // input data (e.g. set-points)
+#define ID_OUTPUT   0x70        // output data (e.g. measurement values)
+#define ID_REC      0xA0        // recorded data (history-dependent)
+#define ID_CAL      0xD0        // calibration
+#define ID_EXEC     0xE0        // function call
+#define ID_AUTH     0xEA
+#define ID_PUB      0xF0        // publication setup
+#define ID_SUB      0xF1        // subscription setup
+#define ID_LOG      0x100       // access log data
+
+#define STRING_LEN 128          // size of allocated strings in config
+#define CONFIG_NODE_NAME        "conf"
+#define CONFIG_NODE_GENERAL     "general"
+#define CONFIG_NODE_EMONCMS     "emoncms"
+#define CONFIG_NODE_MQTT        "mqtt"
```

Comment:
Prefix should be changed also here. Maybe replace with `CONF_`? Or why not keep previous naming "data node" as in charge controller?

Reply:
Yes good point, it turned out as a normal thingset setup so its fine :)